LuckySpin/Controllers/SpinnerController.cs

```diff
@@ -98,12 +102,15 @@ public IActionResult Spin(long id)
          public IActionResult LuckList(long id)
         {
             //Gets the Player belonging to the given id
-            //TODO: Modify the code to use the SingleOrDefault Lamda Extension method
-            Player player = _dbc.Players.Find(id);
+            // Modify the code to use the SingleOrDefault Lamda Extension method
+            Player player = _dbc.Players.SingleOrDefault(p => p.Id == id);
+            
+
             //Gets the list of Spins from the Context
-            //TODO: Modify the next line to get the list of the Player's Spins instead of all the Spins
-            IEnumerable<Spin> spins = _dbc.Spins;
-            // Hack in some detail about the player
+            // Modify the next line to get the list of the Player's Spins instead of all the Spins
+            IEnumerable<Spin> spins = _dbc.Spins.FromSqlInterpolated($"SELECT * FROM dbo.Spins WHERE PlayerID={player.Id}");
```

Comment:
The entity model can do this work for you - take a look at the navigation property for Player.
`player.Spins` is the collection of THAT player's spins